soc: st: stm32: stm32wbax: STM32WBA Cube 1.4.1 FW integration

asm5878 opened the pull request:
STM32WBA Cube FW 1.4.1 integration

Added new functions required by STM32WBA Cube FW 1.4.1 Link Layer for WFI notification management and clock source selection

asm5878 commented on soc/st/stm32/stm32wbax/hci_if/linklayer_plat.c:
```diff
@@ -30,6 +30,10 @@ typedef void (*radio_isr_cb_t) (void);
 radio_isr_cb_t low_isr_callback;
 
+/* Radio bus clock control variables */
+uint8_t AHB5_SwitchedOff = 0;
+uint32_t radio_sleep_timer_val = 0;
+
```
Just retrieving code form Cube FW....

erwango commented on soc/st/stm32/stm32wbax/hci_if/linklayer_plat.c:
```diff
@@ -314,3 +321,27 @@ void LINKLAYER_PLAT_SCHLDR_TIMING_UPDATE_NOT(Evnt_timing_t *p_evnt_timing) {}
 
 void LINKLAYER_PLAT_DisableOSContextSwitch(void) {}
+
+void LINKLAYER_PLAT_NotifyWFIEnter(void)
+{
+	/* Check if Radio state will allow the AHB5 clock to be cut */
+
+	/* AHB5 clock will be cut in the following cases:
+	 * - 2.4GHz radio is not in ACTIVE mode (in SLEEP or DEEPSLEEP mode).
+	 * - RADIOSMEN and STRADIOCLKON bits are at 0.
+	 */
+	if ((LL_PWR_GetRadioMode() != LL_PWR_RADIO_ACTIVE_MODE) ||
+	   ((__HAL_RCC_RADIO_IS_CLK_SLEEP_ENABLED() == 0) &&
+	   (LL_RCC_RADIO_IsEnabledSleepTimerClock() == 0))) {
+		AHB5_SwitchedOff = 1;
+	}
+}
```
This is a STM32Cube function implemented using STM32Cube APIs, it should exist in STM32Cube package and matching file should be included as part of https://github.com/zephyrproject-rtos/hal_stm32/tree/main/lib/stm32wba/hci.

This should STM32Cube specific code should be maintained outside of Zephyr and forked.
Risk of including such code is that maintenance will be complex as we can't get code evolutions.
Also, it raises License questions.

Same for functions introduced in soc/st/stm32/stm32wbax/hci_if/ll_sys_if.c.

asm5878 replied:
A better separation has been done to move STM32CUBE API based code inside  https://github.com/zephyrproject-rtos/hal_stm32/tree/main/lib/stm32wba/hci

asm5878 committed:
soc: st: stm32: stm32wbax: Files renaming

Files renaming done to better isolate zephyr related
functions from stm32 hal related functions

Signed-off-by: Alessandro Manganaro <alessandro.manganaro@st.com>

asm5878 committed:
soc: st: stm32: stm32wbax: STM32WBA Cube 1.4.1 integration

Removed unnecessary pure HAL stm32 functions

Headers cleanup

Signed-off-by: Alessandro Manganaro <alessandro.manganaro@st.com>

erwango commented on soc/st/stm32/stm32wbax/hci_if/linklayer_plat_adapt.c:
```diff
@@ -207,11 +165,6 @@ void LINKLAYER_PLAT_DisableIRQ(void)
 	irq_counter++;
 }
 
-void LINKLAYER_PLAT_Assert(uint8_t condition)
-{
-	__ASSERT_NO_MSG(condition);
-}
-
```
This should be kept.
It allows to generate Zephyr ASSERTS

asm5878 replied:
Fixed

erwango commented on soc/st/stm32/stm32wbax/hci_if/linklayer_plat_adapt.c:
```diff
@@ -4,22 +4,14 @@
-#include <zephyr/irq.h>
 #include <zephyr/kernel.h>
-#include <zephyr/arch/cpu.h>
-#include <zephyr/sys/util.h>
 #include <zephyr/drivers/entropy.h>
 #include <zephyr/logging/log.h>
```
Are you sure file will compile w/o this ?

asm5878 replied:
Yes, It compiles

erwango reviewed (dismissed):
Nice cleanup.
